### PR TITLE
Make it possible to include credentials on all requests.

### PR DESCRIPTION
**Pull request by Sojborg**
When using a proxy or cdn like AWS cloudfront you might need to pass cookies when doing http requests from your web application to your backend service. Ex if your infrastructure requires a token cookie to pass requests through to the backend endpoints.

Right now the oidc-client-ts library doesn't support using credentials on all fetch calls. Because of this the library doesn't support the above use case.

I looked at this pull request: [https://github.com/authts/oidc-client-ts/pull/659](https://github.com/authts/oidc-client-ts/pull/659) which adds support of credentials for the refreshToken fetch request. But it is only supports refreshToken.

In my pull request I changed this to be a fetchRequestCredentials which will put credentials on all fetch calls within the library because in the above use case credentials in needed in all fetch requests.

I could also add support credentials for each fetch request like the refreshToken so there would be settings like this:
- userInfoCredentials
- refreshTokenCredentials
- tokenCredentials
- openConfigurationCredentials
- etc

What is your feedback on this?

### Checklist

- [x] This PR makes changes to the public API 
- [x] I have included links for closing relevant issue numbers

### CODE REVIEW

**Comment by pamapa** [comment]
thanks for contributing, i will merge when all issues are resolved...

File: dist/esm/package.json [hunk 1]
```diff
@@ -1,3 +0,0 @@
-{
-  "type": "module"
-}
```
Comment (pamapa) [inline dist/esm/package.json:3]: please leave this file as is, if you really want to remove it, make a separate MR with this...

File: src/UserInfoService.ts [hunk 2]
```diff
@@ -13,7 +14,8 @@ export class UserInfoService {
     protected readonly _logger = new Logger("UserInfoService");
     private readonly _jsonService: JsonService;
 
-    public constructor(private readonly _metadataService: MetadataService) {
+    public constructor(private readonly _metadataService: MetadataService,
+        private readonly _settings: OidcClientSettingsStore) {
```
Comment (pamapa) [inline src/UserInfoService.ts:18]: cosmetics: please move _settings before _metadataService parameter

File: src/OidcClientSettings.ts [hunk 3]
```diff
@@ -104,14 +104,15 @@ export interface OidcClientSettings {
     extraTokenParams?: Record<string, unknown>;
 
     /**
-     * Credentials used by fetch with the refresh request. (default: "same-origin")
+     * Will check the content type header of the response of the revocation endpoint to match these passed values (default: [])
      */
-    refreshTokenCredentials?: "same-origin" | "include" | "omit";
+    revokeTokenAdditionalContentTypes?: string[];
```
Comment (pamapa) [inline src/OidcClientSettings.ts:109]: 1. please do not change the sequence, it makes it hard to see the diff
2. the rename refreshTokenCredentials to fetchRequestCredentials is a breaking change, maybe its better to leave this as is or we need to keep it and add the new one and merge them with this file. We can then also deprecate the old one...

Comment (Sojborg) [reply src/OidcClientSettings.ts:109]: Okay, so I keep the refreshTokenCredentials even though i removed all usages of it in the code base? I get the idea I just want to make sure that i understand it correctly.

File: dist/esm/package.json (hunk 1 above)
Comment (Sojborg) [reply dist/esm/package.json:3]: Yeah, I think i just deleted the whole dist library because i thought it was auto generated :). Will add it back in.

**Comment by Sojborg** [comment]
Thank you for fast replies. Pushed the fixes. Just have a minor question around refreshTokenCredentials.

File: src/OidcClientSettings.ts (hunk 3 above)
Comment (pamapa) [reply src/OidcClientSettings.ts:114]: yes, in src/OidcClientSettings.ts we then take the new `fetchRequestCredentials` if defined, else we fall-back to the deprecated `refreshTokenCredentials`

something like this
```
this.fetchRequestCredentials = fetchRequestCredentials ? fetchRequestCredentials
    : refreshTokenCredentials ? refreshTokenCredentials : "same-origin";
```

Nice to have: if both are defined print a warning or throw an Error. Not sure if worth...

Comment (Sojborg) [reply src/OidcClientSettings.ts:114]: Ah yes of cause to still support the refreshToken. I pushed the changes. I think it makes sense with a warn if both properties are set. An error is a bit aggressive as the functionality will still work.

File: src/MetadataService.ts [hunk 4]
```diff
@@ -17,6 +17,7 @@ export class MetadataService {
     private _metadataUrl: string;
     private _signingKeys: SigningKey[] | null = null;
     private _metadata: Partial<OidcMetadata> | null = null;
+    private fetchRequestCredentials: RequestCredentials | undefined;
```
Comment (pamapa) [inline src/MetadataService.ts:20]: `fetchRequestCredentials` -> `_fetchRequestCredentials` (underscore)

Comment (pamapa) [reply src/MetadataService.ts:20]: and align type and default value to others: 
`private _fetchRequestCredentials: RequestCredentials | null = null;`

Comment (Sojborg) [reply src/MetadataService.ts:20]: Default value should be undefined then the credentials property will skipped if the _fetchRequestCredentials isn't set.

Comment (Sojborg) [reply src/MetadataService.ts:20]: Ex here: https://github.com/authts/oidc-client-ts/pull/778/files#diff-dc78e3cc058d3fd2461eec10c5b75ee1c28394a27aadf3ab8a8fe3294547b9e3R58

Comment (Sojborg) [reply src/MetadataService.ts]: Renaming done.

Comment (pamapa) [reply src/MetadataService.ts]: so only the rename (fetchRequestCredentials -> _fetchRequestCredentials) here...

**Review by pamapa — APPROVED** [review]

**Comment by Sojborg** [comment]
And thank you for being fast at replying. Really nice 🥳

**Comment by Sojborg** [comment]
@pamapa when will you make a new npm version with the changes?

**Comment by pamapa** [comment]
today :-)
